lib/rspec/core/memoized_helpers.rb

```diff
@@ -123,14 +123,28 @@ def is_expected
 
       # @private
       def __memoized
-        @__memoized ||= {}
+        @__memoized ||= Memoized.new
```

Comment:
`||=` isn't threadsafe.  Could there be a race condition when `__memoized` is accessed in parallel threads before the ivar has been initialized?